packages/core/src/indexing/local-index-api.ts

```diff
@@ -198,6 +209,17 @@ export class LocalIndexApi implements IndexApi {
     // back to the DatabaseIndexApi so that it can populate its internal state.
     // TODO(CDB-2132):  Fix this fragile and circular DatabaseApi initialization
     const modelsToIndex = this.databaseIndexApi.getIndexedModels()
+    for (const model of modelsToIndex) {
```

Comment:
can we make a helper function for this type checking?  Even better would be to use a codec, but maybe that's better saved for a follow-up to not hold up this PR any more